scripts/cxx-interop-compatibility.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+mkdir -p interop
```

Comment:
Would be good if we can use temporary directory instead to avoid collisions and not clutter the checkout if this is run locally, for example.